src/shared/test/port-util-test.js

```diff
@@ -0,0 +1,140 @@
+import { isMessageEqual } from '../port-util';
+
+const source = 'hypothesis';
+
+describe('port-util', () => {
+  describe('isMessageEqual', () => {
+    [
+      {
+        data: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        message: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        expectedResult: true,
+        reason: 'data matches the message',
+      },
+      {
+        data: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        message: {
+          source,
+          type: 'offer',
+          channel: 'host-sidebar',
+          port: 'guest',
+        },
+        expectedResult: true,
+        reason: 'data matches the message (properties in different order)',
+      },
+      {
+        data: null,
+        message: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        expectedResult: false,
+        reason: 'data is null',
+      },
+      {
+        data: {
+          channel: 'host-sidebar',
+          port: 9, // wrong type
+          source,
+          type: 'offer',
+        },
+        message: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        expectedResult: false,
+        reason: 'data has a property with the wrong type',
+      },
+      {
+        data: {
+          // channel property missing
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        message: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        expectedResult: false,
+        reason: 'data has one property that is missing',
+      },
+      {
+        data: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+          extra: 'dummy', // additional
+        },
+        message: {
+          channel: 'host-sidebar',
+          port: 'guest',
+          source,
+          type: 'offer',
+        },
+        expectedResult: false,
+        reason: 'data has one additional property',
```

Comment:
I wonder whether we might ever see a situation where the browser or third-party code modifies messages and adds extra properties. I guess we'll find out when we deploy it.